martin/src/srv/server.rs

```diff
@@ -79,8 +80,9 @@ async fn get_health() -> impl Responder {
     wrap = "middleware::Compress::default()"
 )]
 #[allow(clippy::unused_async)]
-async fn get_catalog(catalog: Data<Catalog>) -> impl Responder {
-    HttpResponse::Ok().json(catalog)
+async fn get_catalog(catalog: Data<Arc<RwLock<Catalog>>>) -> impl Responder {
+    let catalog_read = catalog.read().expect("Failed to acquire read lock on catalog");
+    HttpResponse::Ok().json(&*catalog_read)
```

Comment:
in Rust, people tend to re-use the same var name.  Also, please enable `cargo fmt` on save in your IDE.  E.g. [vscode](https://stackoverflow.com/questions/67859926/how-to-run-cargo-fmt-on-save-in-vscode) or [intellij](https://stackoverflow.com/questions/59512214/execute-rustfmt-on-file-save-in-intellij-or-clion)

```suggestion
    let catalog = catalog
        .read()
        .expect("Failed to acquire read lock on catalog");
    HttpResponse::Ok().json(&*catalog)
```